util/math_reference.h

```diff
@@ -305,21 +305,30 @@ sycl_cts::resultRef<sycl::marray<T, N>> abs(sycl::marray<T, N> a) {
 
 /* absolute difference */
 template <typename T>
-T abs_diff(T a, T b) {
+sycl_cts::resultRef<T> abs_diff(T a, T b) {
+  using U = std::make_unsigned_t<T>;
   T h = (a > b) ? a : b;
   T l = (a <= b) ? a : b;
-  return h - l;
+  // Using two's-complement and that unsigned integer underflow is defined as
+  // modulo 2^n we get the result by computing the distance based on signed
```

Comment:
Thanks, now I understand the code. :-)
But is it signed? It looks like it uses unsigned comparison.